src/main/app/src/app/shared/pipes/datum.pipe.ts

```diff
@@ -6,45 +6,61 @@
 import { Inject, LOCALE_ID, Pipe, PipeTransform } from "@angular/core";
 import moment from "moment";
 
+type DateFormat = readonly [
+  "shortDate",
+  "mediumDate",
+  "longDate",
+  "short",
+  "medium",
+  "long",
+  "full",
+  "fullDate",
+];
```

Comment:
Making this type a union string literal would make it easier to read

```suggestion
type DateFormat =
  | "shortDate"
  | "mediumDate"
  | "longDate"
  | "short"
  | "medium"
  | "long"
  | "full"
  | "fullDate";
```

Of course in other places needs slight changes as well

Reply:
Agreed, that was my first attempt - however, then you lose the extra type hints in your `HTML` templates unfortuantly.

Reply:
okay weird, I tried it again for my own sanity and it does work in the template now. ts-server must have been lazy yesterday!